keywrap/pkcs11/keywrapper_pkcs11.go

```diff
@@ -0,0 +1,233 @@
+/*
+   Copyright The ocicrypt Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package pkcs11
+
+import (
+	"github.com/containers/ocicrypt/config"
+	"github.com/containers/ocicrypt/keywrap"
+	"github.com/miekg/pkcs11"
+	"github.com/pkg/errors"
+	"strings"
+)
+
+var (
+	KeyLabel   = "imgenc"
+	OAEPParams = &pkcs11.OAEPParams{
+		HashAlg:    pkcs11.CKM_SHA_1,
+		MGF:        pkcs11.CKG_MGF1_SHA1,
+		SourceType: pkcs11.CKZ_DATA_SPECIFIED,
+		SourceData: nil,
+	}
+)
+
+type pkcs11KeyWrapper struct {
+}
+
+func NewKeyWrapper() keywrap.KeyWrapper {
+	return &pkcs11KeyWrapper{}
+}
+
+func (kw pkcs11KeyWrapper) GetAnnotationID() string {
+	return "org.opencontainers.image.enc.keys.experimental.pkcs11"
+}
+
+func (kw pkcs11KeyWrapper) WrapKeys(ec *config.EncryptConfig, optsData []byte) ([]byte, error) {
+	if len(ec.Parameters["modules"]) == 0 || len(ec.Parameters["pins"]) == 0 {
```

Comment:
nit, but could we use the subcall NoPossibleKeys for this check>?

Reply:
fixed.